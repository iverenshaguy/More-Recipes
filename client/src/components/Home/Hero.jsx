import React from 'react';
import { InputGroup, InputGroupButton, Button, Form, Label, Input } from 'reactstrap';
import FontAwesome from 'react-fontawesome';
import PropTypes from 'prop-types';

const Hero = props => (
  <div className="container-fluid" id="hero-image">
    <div className="row mx-auto" id="hero-image-text">
      <div className="col align-self-center">
        <div className="row justify-content-center align-items-center mb-3">
          <div className="col-12">
            <h3>Search for a Recipe</h3>
          </div>
          <div className="col-8 py-4">
            <Form id="search-form">
              <InputGroup size="lg">
                <Label className="sr-only" for="search">
                  Search for Recipes
                </Label>
                <Input id="search" placeholder="Curried Chicken Gravy..." />
                <InputGroupButton>
                  <Button type="submit" className="search-btn">
                    <FontAwesome name="search" />
                  </Button>
                </InputGroupButton>
              </InputGroup>
            </Form>
          </div>
        </div>
        <Button
          onClick={() => props.toggle('addRecipeModal')}
          className="btn-default btn-lg"
          id="home-add-recipe-btn"
          title="New Recipe"
        >
          Add a New Recipe
        </Button>
      </div>
    </div>
  </div>
);

Hero.propTypes = {
  toggle: PropTypes.func.isRequired,
};

export default Hero;
