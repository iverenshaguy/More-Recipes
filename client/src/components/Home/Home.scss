@import '../../assets/scss/modules';

.home {
  .navbar {
    position: relative;
  }

  #hero-image {
    height: 45vh;
    max-width: 100%;
    position: relative;

    button {
      background-color: $light-green;
      border: 1px solid $light-green;
      color: $white;
      padding: 0.75rem 1.5rem;

      &:hover,
      &:focus,
      &:active {
        background-color: $green;
        border: 1px solid $green;
        color: $white;
      }
    }
  }

  #hero-image-text {
    color: $white;
    padding-top: 5vh;
    text-align: center;
    width: 100%;
  }

  #hero-search-form {
    margin-top: 2rem;
  }
}

#home-add-recipe-btn {
  border-radius: 0;
  box-shadow: 0;
  outline: none;

  &:hover,
  &:focus,
  &:active {
    cursor: pointer;
  }
}
