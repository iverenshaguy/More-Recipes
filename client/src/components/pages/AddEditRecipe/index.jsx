import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Form from '../../shared/Forms';
import { urlMatchPropTypes } from '../../../helpers/proptypes';

/**
 * @exports
 * @class AddEditRecipe
 * @extends Component
 * @returns {component} AddEditRecipe
 */
class AddEditRecipe extends Component {
  static propTypes = {
    ...urlMatchPropTypes,
    type: PropTypes.string,
    uploading: PropTypes.bool.isRequired,
    submitting: PropTypes.bool.isRequired,
  };

  static defaultProps = {
    type: 'addRecipe'
  }

  /**
   * @memberof AddEditRecipe
   * @returns {component} AddEditRecipe
   */
  render() {
    const meta = {
      title: this.props.type === 'addRecipe' ? 'Add a New Recipe' : 'Edit Recipe',
      btnText: this.props.type === 'addRecipe' ? 'Add Recipe' : 'Edit Recipe'
    };

    return (
      <div className="card-form-body">
        <div className="container">
          <div className="my-5" id="card-form-wrapper">
            <div className="p-xs-2 pt-4 pb-3" id="login-signup-div">
              <Form {...this.props} type={this.props.type} meta={meta} />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

const mapStateToProps = state => ({
  submitting: state.singleRecipe.recipe.adding,
  uploading: state.uploadImage.uploading,
  submitError: state.singleRecipe.recipe.error
});

export { AddEditRecipe as AddEditRecipeComponent };
export default connect(mapStateToProps)(AddEditRecipe);
