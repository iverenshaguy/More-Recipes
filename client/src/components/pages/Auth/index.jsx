import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Link, Redirect } from 'react-router-dom';
import { Row, Col } from 'reactstrap';
import Form from '../../shared/Forms';
import setCurrentLocation from '../../../actions/location';
import { authPropTypes } from '../../../helpers/proptypes';
import './Auth.scss';

/**
 * @exports
 * @class Auth
 * @extends Component
 * @returns {component} Auth
 */
class Auth extends Component {
  static propTypes = {
    ...authPropTypes,
    type: PropTypes.string.isRequired,
    dispatch: PropTypes.func.isRequired,
  };

  /**
   * @memberof LoginForm
   * @returns {nothing} Returns nothing
   */
  componentWillMount() {
    this.props.dispatch(setCurrentLocation('auth'));
  }

  /**
   * @memberof Auth
   * @returns {component} Auth
   */
  render() {
    const { from } = this.props.location.state ? this.props.location.state : { from: { pathname: '/' } };

    if (this.props.isAuthenticated) {
      return <Redirect to={from} />;
    }

    const signupMeta = {
      title: 'Register for a New Account',
      btnText: 'SIGN UP',
      extra: <p className="text-center">Already have an account, <Link to="/login">Log in here</Link>.</p>
    };

    const loginMeta = {
      title: 'Sign In to Your Account',
      btnText: 'SIGN IN',
      extra: <p className="text-center">Dont have an account, <Link to="/signup">Sign up here</Link>.</p>
    };

    return (
      <div className="login-signup-body">
        <div className="container" id="login-signup-container">
          <Row className="my-5" id="login-signup-wrapper">
            <Col className="p-xs-2 p-lg-4" id="login-signup-div">
              {this.props.type === 'signup' ?
                <Form {...this.props} type="signup" meta={signupMeta} /> :
                <Form {...this.props} type="login" meta={loginMeta} />}
            </Col>
          </Row>
        </div>
      </div>
    );
  }
}

export { Auth as AuthComponent };

const mapStateToProps = state => ({
  submitting: state.auth.loading,
  isAuthenticated: state.auth.isAuthenticated,
  submitError: state.auth.error
});

export default connect(mapStateToProps)(Auth);
