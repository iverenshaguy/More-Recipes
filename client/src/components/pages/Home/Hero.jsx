import React from 'react';
import PropTypes from 'prop-types';
import { AddRecipeBtn } from '../../shared/Buttons';
import SearchForm from '../../shared/Forms/SearchForm';

/**
 * @exports
 * @function Hero
 * @param {object} props
 * @returns {JSX} Hero
 */
const Hero = (props) => {
  const {
    handleSearch, handleAddRecipe, handleSearchInput, searchValue
  } = props;

  return (
    <div className="container-fluid" id="hero-image">
      <div className="row mx-auto" id="hero-image-text">
        <div className="col align-self-center">
          <div className="row justify-content-center align-items-center mb-3">
            <div className="col-12">
              <h3>Search for a Recipe</h3>
            </div>
            <div className="col-8 py-4">
              <SearchForm
                size="lg"
                handleSearch={handleSearch}
                handleSearchInput={handleSearchInput}
                searchValue={searchValue}
              />
            </div>
          </div>
          <AddRecipeBtn handleClick={handleAddRecipe} />
        </div>
      </div>
    </div>);
};

Hero.propTypes = {
  handleSearch: PropTypes.func.isRequired,
  handleAddRecipe: PropTypes.func.isRequired,
  handleSearchInput: PropTypes.func.isRequired,
  searchValue: PropTypes.string.isRequired
};

export default Hero;
