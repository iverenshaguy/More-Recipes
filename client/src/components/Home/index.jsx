import React, { Component } from 'react';
import PropTypes from 'prop-types';
import Hero from './Hero';
import RecipeItems from '../shared/RecipeItems';
import './Home.scss';

/**
 * @exports
 * @class Home
 * @extends Component
 * @returns {component} Home
 */
class Home extends Component {
  /**
   * @memberof Home
   * @returns {function} updateLocationState
   */
  componentWillMount() {
    this.props.updateLocationState('home');
  }

  /**
   * @memberof Home
   * @returns {component} Home
   */
  render() {
    return (
      <div className="home">
        <Hero toggle={this.props.toggleModal} />
        <RecipeItems />
      </div>
    );
  }
}

Home.propTypes = {
  toggleModal: PropTypes.func.isRequired,
  updateLocationState: PropTypes.func.isRequired,
};

export default Home;
