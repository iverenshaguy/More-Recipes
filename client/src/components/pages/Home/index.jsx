import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import setCurrentLocation from '../../../actions/location';
import { fetchTopRecipes, searchRecipes } from '../../../actions/recipes';
import Hero from './Hero';
import RecipeItems from '../../shared/RecipeItems';
import PreLoader from '../../shared/PreLoader';
import './Home.scss';

/**
 * @exports
 * @class Home
 * @extends Component
 * @returns {component} Home
 */
class Home extends Component {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
    isFetching: PropTypes.bool.isRequired,
    recipes: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.number,
      recipeName: PropTypes.string,
      recipeImage: PropTypes.string,
      prepTime: PropTypes.string,
      cookTime: PropTypes.string,
      totalTime: PropTypes.string,
      difficulty: PropTypes.string,
      extraInfo: PropTypes.string,
      vegetarian: PropTypes.bool,
      ingredients: PropTypes.array,
      preparations: PropTypes.array,
      directions: PropTypes.array,
      upvotes: PropTypes.number,
      downvotes: PropTypes.number,
      views: PropTypes.number,
      createdAt: PropTypes.string,
      updatedAt: PropTypes.string,
      userId: PropTypes.number,
      rating: PropTypes.string
    })).isRequired,
    metaData: PropTypes.shape({
      firstPage: PropTypes.number,
      lastPage: PropTypes.number,
      page: PropTypes.number,
      pageRecipeCount: PropTypes.number,
      pages: PropTypes.arrayOf(PropTypes.number),
      totalRecipeCount: PropTypes.number,
    }).isRequired,
  };

  /**
   * @constructor
   * @memberof Home
   * @returns {nothing} - Returns nothing
   */
  constructor() {
    super();

    this.state = {
      title: 'TOP RECIPES',
      currentPage: 1,
      limit: 5,
      searchValue: ''
    };

    this.handlePageChange = this.handlePageChange.bind(this);
    this.handleSearchInput = this.handleSearchInput.bind(this);
    this.handleSearch = this.handleSearch.bind(this);
  }

  /**
   * @memberof Home
   * @return {nothing} Returns nothing
   */
  componentWillMount() {
    this.props.dispatch(setCurrentLocation('home'));
    this.props.dispatch(fetchTopRecipes(this.state.currentPage, this.state.limit));
  }

  /**
   * @memberof Home
   * @param {number} page
   * @returns {state} home
   */
  handlePageChange(page) {
    let fetch;
    const { limit, searchValue } = this.state;

    if (this.state.title === 'TOP RECIPES') {
      fetch = fetchTopRecipes(page, limit);
    } else {
      fetch = searchRecipes(searchValue, page, limit);
    }

    this.setState({
      currentPage: page
    }, () => this.props.dispatch(fetch));
  }

  /**
   * @memberof Home
   * @param {event} event
   * @returns {state} home
   */
  handleSearchInput(event) {
    this.setState({
      searchValue: event.target.value
    });
  }

  /**
   * @memberof Home
   * @param {object} event
   * @returns {state} home
   */
  handleSearch(event) {
    event.preventDefault();

    const { searchValue, currentPage, limit } = this.state;
    this.setState({
      title: 'SEARCH RESULTS'
    }, () => this.props.dispatch(searchRecipes(searchValue, currentPage, limit)));
  }

  /**
   * @memberof Home
   * @returns {fragment} Recipe items
   */
  renderBody() {
    const { isFetching, recipes, metaData } = this.props;
    const { title } = this.state;

    if (isFetching) {
      return <PreLoader />;
    }

    return (
      <RecipeItems
        title={title}
        recipes={recipes}
        handlePageChange={this.handlePageChange}
        metaData={metaData}
      />
    );
  }

  /**
   * @memberof Home
   * @returns {component} Home
   */
  render() {
    return (
      <div className="home">
        <Hero
          dispatch={this.props.dispatch}
          handleSearchInput={this.handleSearchInput}
          handleSearch={this.handleSearch}
          searchValue={this.state.searchValue}
        />
        {this.renderBody()}
      </div>
    );
  }
}

const mapStateToProps = state => ({
  isFetching: state.isFetching,
  recipes: state.recipes.recipes,
  metaData: state.recipes.metaData
});

export { Home as HomeComponent };

export default connect(mapStateToProps)(Home);
