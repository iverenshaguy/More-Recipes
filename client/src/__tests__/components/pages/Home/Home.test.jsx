import React from 'react';
import { HomeComponent } from '../../../../components/pages/Home';
import setCurrentLocation from '../../../../actions/location';

const setup = () => {
  const dispatch = jest.fn();

  const props = {
    dispatch,
    isFetching: false,
    recipes: [{ id: 1, recipeName: 'Rice' }, { id: 5, recipeName: 'Beans' }],
    metadata: {
      firstPage: 1,
      lastPage: 3,
      page: 2,
      itemsPerPage: 5,
      pages: [1, 2, 3],
      totalCount: 13,
    }
  };

  const state = {
    title: 'TOP RECIPES',
    currentPage: 2,
    limit: 5,
    searchValue: ''
  };

  const mountedWrapper = mount(<HomeComponent {...props} />);
  const shallowWrapper = shallow(<HomeComponent {...props} />);

  return {
    state, props, mountedWrapper, shallowWrapper
  };
};

describe('Home', () => {
  afterAll(() => {
    jest.clearAllMocks();
  });

  it('renders correctly', () => {
    const { shallowWrapper } = setup();
    expect(toJson(shallowWrapper)).toMatchSnapshot();
  });

  it('renders correctly when fully mounted', () => {
    const { mountedWrapper } = setup();
    expect(toJson(mountedWrapper)).toMatchSnapshot();
    mountedWrapper.unmount();
  });

  it('renders preloader if fetching', () => {
    const { props } = setup();
    const mountedWrapper = mount(<HomeComponent {...props} isFetching />);

    expect(mountedWrapper.find('PreLoader').length).toEqual(1);
    mountedWrapper.unmount();
  });

  it('calls component will mount when mounted', () => {
    const { mountedWrapper, props } = setup();
    expect(props.dispatch).toHaveBeenCalledWith(setCurrentLocation('home'));
    mountedWrapper.unmount();
  });

  it('handles page change for TOP RECIPES', () => {
    const { mountedWrapper, state } = setup();

    const handlePageChangeSpy = jest.spyOn(mountedWrapper.instance(), 'handlePageChange');

    mountedWrapper.setState(state);

    mountedWrapper.find('a.page-link').at(4).simulate('click');

    expect(handlePageChangeSpy).toHaveBeenCalled();
    expect(mountedWrapper.instance().state.currentPage).toEqual(3);

    mountedWrapper.unmount();
  });

  it('handles page change for SEARCH RECIPES', () => {
    const { mountedWrapper, state } = setup();

    const handlePageChangeSpy = jest.spyOn(mountedWrapper.instance(), 'handlePageChange');

    mountedWrapper.setState({ ...state, title: 'SEARCH RESULTS' });

    mountedWrapper.find('a.page-link').at(3).simulate('click');

    expect(handlePageChangeSpy).toHaveBeenCalled();
    expect(mountedWrapper.instance().state.currentPage).toEqual(2);

    mountedWrapper.unmount();
  });

  it('handles search input', () => {
    const { mountedWrapper, props, state } = setup();

    const event = { target: { value: 'Chicken' } };

    mountedWrapper.setState(state);
    mountedWrapper.find('input#search').simulate('change', event);

    expect(props.dispatch).toHaveBeenCalled();
    expect(mountedWrapper.instance().state.searchValue).toEqual('Chicken');

    mountedWrapper.unmount();
  });

  it('handles search', () => {
    const { mountedWrapper, props, state } = setup();

    const event = { target: { value: 'Chicken' } };

    mountedWrapper.setState(state);
    mountedWrapper.find('input#search').simulate('change', event);
    mountedWrapper.find('form').simulate('submit', { preventDefault() { } });

    expect(props.dispatch).toHaveBeenCalled();
    expect(mountedWrapper.instance().state.title).toEqual('SEARCH RESULTS');

    mountedWrapper.unmount();
  });
});
