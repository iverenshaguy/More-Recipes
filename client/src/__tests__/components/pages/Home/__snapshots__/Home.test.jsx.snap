// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  className="home"
>
  <Hero
    dispatch={[Function]}
    handleSearch={[Function]}
    handleSearchInput={[Function]}
    searchValue=""
  />
  <RecipeItems
    handlePageChange={[Function]}
    metaData={
      Object {
        "firstPage": 1,
        "lastPage": 3,
        "page": 2,
        "pageRecipeCount": 5,
        "pages": Array [
          1,
          2,
          3,
        ],
        "totalRecipeCount": 13,
      }
    }
    recipes={
      Array [
        Object {
          "id": 1,
          "recipeName": "Rice",
        },
        Object {
          "id": 5,
          "recipeName": "Beans",
        },
      ]
    }
    title="TOP RECIPES"
  />
</div>
`;

exports[`Home renders correctly when fully mounted 1`] = `
<Home
  dispatch={[Function]}
  isFetching={false}
  metaData={
    Object {
      "firstPage": 1,
      "lastPage": 3,
      "page": 2,
      "pageRecipeCount": 5,
      "pages": Array [
        1,
        2,
        3,
      ],
      "totalRecipeCount": 13,
    }
  }
  recipes={
    Array [
      Object {
        "id": 1,
        "recipeName": "Rice",
      },
      Object {
        "id": 5,
        "recipeName": "Beans",
      },
    ]
  }
>
  <div
    className="home"
  >
    <Hero
      dispatch={[Function]}
      handleSearch={[Function]}
      handleSearchInput={[Function]}
      searchValue=""
    >
      <div
        className="container-fluid"
        id="hero-image"
      >
        <div
          className="row mx-auto"
          id="hero-image-text"
        >
          <div
            className="col align-self-center"
          >
            <div
              className="row justify-content-center align-items-center mb-3"
            >
              <div
                className="col-12"
              >
                <h3>
                  Search for a Recipe
                </h3>
              </div>
              <div
                className="col-8 py-4"
              >
                <Form
                  id="search-form"
                  onSubmit={[Function]}
                  tag="form"
                >
                  <form
                    className=""
                    id="search-form"
                    onSubmit={[Function]}
                  >
                    <InputGroup
                      size="lg"
                      tag="div"
                    >
                      <div
                        className="input-group input-group-lg"
                      >
                        <Label
                          className="sr-only"
                          for="search"
                          tag="label"
                          widths={
                            Array [
                              "xs",
                              "sm",
                              "md",
                              "lg",
                              "xl",
                            ]
                          }
                        >
                          <label
                            className="sr-only  form-control-label"
                            htmlFor="search"
                          >
                            Search for Recipes
                          </label>
                        </Label>
                        <Input
                          id="search"
                          onChange={[Function]}
                          placeholder="Curried Chicken Gravy..."
                          tag="p"
                          type="text"
                          value=""
                        >
                          <input
                            className="form-control"
                            id="search"
                            onChange={[Function]}
                            placeholder="Curried Chicken Gravy..."
                            type="text"
                            value=""
                          />
                        </Input>
                        <InputGroupButton
                          tag="div"
                        >
                          <div
                            className="input-group-btn"
                          >
                            <Button
                              className="search-btn"
                              color="secondary"
                              tag="button"
                              type="submit"
                            >
                              <button
                                className="search-btn btn btn-secondary"
                                onClick={[Function]}
                                type="submit"
                              >
                                <FontAwesome
                                  name="search"
                                >
                                  <span
                                    aria-hidden={true}
                                    className="fa fa-search"
                                  />
                                </FontAwesome>
                              </button>
                            </Button>
                          </div>
                        </InputGroupButton>
                      </div>
                    </InputGroup>
                  </form>
                </Form>
              </div>
            </div>
            <Button
              className="btn-default btn-lg"
              color="secondary"
              id="home-add-recipe-btn"
              onClick={[Function]}
              tag="button"
              title="New Recipe"
            >
              <button
                className="btn-default btn-lg btn btn-secondary"
                id="home-add-recipe-btn"
                onClick={[Function]}
                title="New Recipe"
                type="button"
              >
                Add a New Recipe
              </button>
            </Button>
          </div>
        </div>
      </div>
    </Hero>
    <RecipeItems
      handlePageChange={[Function]}
      metaData={
        Object {
          "firstPage": 1,
          "lastPage": 3,
          "page": 2,
          "pageRecipeCount": 5,
          "pages": Array [
            1,
            2,
            3,
          ],
          "totalRecipeCount": 13,
        }
      }
      recipes={
        Array [
          Object {
            "id": 1,
            "recipeName": "Rice",
          },
          Object {
            "id": 5,
            "recipeName": "Beans",
          },
        ]
      }
      title="TOP RECIPES"
    >
      <div
        className="container-fluid pb-5 background-grey page-margin"
        id="search-result-wrapper"
      >
        <div
          className="heading pt-4 text-center"
          id="search-result-heading"
        >
          <div
            className="col-xs-12 col-md-12 pt-3"
            id="title"
          >
            <h5>
              TOP RECIPES
            </h5>
          </div>
        </div>
        <hr />
        <div
          className="row pt-2 px-5"
          id="search-result"
        >
          <RecipeCard
            key="1"
            recipe={
              Object {
                "id": 1,
                "recipeName": "Rice",
              }
            }
          >
            <div
              className="col-lg-3 col-md-6 col-sm-12 col-xs-12 mt-4 recipe-card"
            >
              <Card
                tag="div"
              >
                <div
                  className="card"
                >
                  <a
                    href="view-recipe.html"
                  >
                    <CardImg
                      alt="recipe"
                      className="img-fluid"
                      src="images/jollof-rice-img.jpg"
                      tag="img"
                      top={true}
                      width="100%"
                    >
                      <img
                        alt="recipe"
                        className="img-fluid card-img-top"
                        src="images/jollof-rice-img.jpg"
                        width="100%"
                      />
                    </CardImg>
                    <CardBody
                      className="card-block text-center pb-3"
                      tag="div"
                    >
                      <div
                        className="card-block text-center pb-3 card-body"
                      >
                        <h5
                          className="mb-1"
                        >
                          Rice
                        </h5>
                      </div>
                    </CardBody>
                    <CardBody
                      className="card-block text-center recipe-details"
                      tag="div"
                    >
                      <div
                        className="card-block text-center recipe-details card-body"
                      >
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              name="clock-o"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-clock-o"
                              />
                            </FontAwesome>
                             
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-0 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-0 d-inline card-text"
                          >
                             | 
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted d-inline card-text"
                          >
                            <i
                              className="aria-hidden flaticon flaticon-oven-kitchen-tool-for-cooking-foods"
                            />
                          </p>
                        </CardText>
                      </div>
                    </CardBody>
                    <CardBody
                      className="text-center text-muted px-2 recipe-rating"
                      tag="div"
                    >
                      <div
                        className="text-center text-muted px-2 recipe-rating card-body"
                      >
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              name="star"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-star"
                              />
                            </FontAwesome>
                             
                            NaN
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                             | 
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              name="heart"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-heart"
                              />
                            </FontAwesome>
                             
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                             | 
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              className="text-success"
                              name="eye"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-eye text-success"
                              />
                            </FontAwesome>
                             
                          </p>
                        </CardText>
                      </div>
                    </CardBody>
                  </a>
                </div>
              </Card>
            </div>
          </RecipeCard>
          <RecipeCard
            key="5"
            recipe={
              Object {
                "id": 5,
                "recipeName": "Beans",
              }
            }
          >
            <div
              className="col-lg-3 col-md-6 col-sm-12 col-xs-12 mt-4 recipe-card"
            >
              <Card
                tag="div"
              >
                <div
                  className="card"
                >
                  <a
                    href="view-recipe.html"
                  >
                    <CardImg
                      alt="recipe"
                      className="img-fluid"
                      src="images/jollof-rice-img.jpg"
                      tag="img"
                      top={true}
                      width="100%"
                    >
                      <img
                        alt="recipe"
                        className="img-fluid card-img-top"
                        src="images/jollof-rice-img.jpg"
                        width="100%"
                      />
                    </CardImg>
                    <CardBody
                      className="card-block text-center pb-3"
                      tag="div"
                    >
                      <div
                        className="card-block text-center pb-3 card-body"
                      >
                        <h5
                          className="mb-1"
                        >
                          Beans
                        </h5>
                      </div>
                    </CardBody>
                    <CardBody
                      className="card-block text-center recipe-details"
                      tag="div"
                    >
                      <div
                        className="card-block text-center recipe-details card-body"
                      >
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              name="clock-o"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-clock-o"
                              />
                            </FontAwesome>
                             
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-0 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-0 d-inline card-text"
                          >
                             | 
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted d-inline card-text"
                          >
                            <i
                              className="aria-hidden flaticon flaticon-oven-kitchen-tool-for-cooking-foods"
                            />
                          </p>
                        </CardText>
                      </div>
                    </CardBody>
                    <CardBody
                      className="text-center text-muted px-2 recipe-rating"
                      tag="div"
                    >
                      <div
                        className="text-center text-muted px-2 recipe-rating card-body"
                      >
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              name="star"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-star"
                              />
                            </FontAwesome>
                             
                            NaN
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                             | 
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              name="heart"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-heart"
                              />
                            </FontAwesome>
                             
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                             | 
                          </p>
                        </CardText>
                        <CardText
                          className="text-muted mr-2 d-inline"
                          tag="p"
                        >
                          <p
                            className="text-muted mr-2 d-inline card-text"
                          >
                            <FontAwesome
                              className="text-success"
                              name="eye"
                            >
                              <span
                                aria-hidden={true}
                                className="fa fa-eye text-success"
                              />
                            </FontAwesome>
                             
                          </p>
                        </CardText>
                      </div>
                    </CardBody>
                  </a>
                </div>
              </Card>
            </div>
          </RecipeCard>
        </div>
        <Pagination
          items={
            Array [
              Object {
                "id": 1,
                "recipeName": "Rice",
              },
              Object {
                "id": 5,
                "recipeName": "Beans",
              },
            ]
          }
          metaData={
            Object {
              "firstPage": 1,
              "lastPage": 3,
              "page": 2,
              "pageRecipeCount": 5,
              "pages": Array [
                1,
                2,
                3,
              ],
              "totalRecipeCount": 13,
            }
          }
          onPageChange={[Function]}
        >
          <div
            className="mt-5"
          >
            <nav
              aria-label="Page navigation"
            >
              <Pagination
                className="justify-content-center"
                tag="ul"
              >
                <ul
                  className="justify-content-center pagination"
                >
                  <PaginationItem
                    className=""
                    tag="li"
                  >
                    <li
                      className="page-item"
                    >
                      <PaginationLink
                        onClick={[Function]}
                        tag="a"
                      >
                        <a
                          className="page-link"
                          onClick={[Function]}
                        >
                          First
                        </a>
                      </PaginationLink>
                    </li>
                  </PaginationItem>
                  <PaginationItem
                    className=""
                    tag="li"
                  >
                    <li
                      className="page-item"
                    >
                      <PaginationLink
                        onClick={[Function]}
                        tag="a"
                      >
                        <a
                          className="page-link"
                          onClick={[Function]}
                        >
                          <span
                            aria-hidden="true"
                          >
                            «
                          </span>
                          <span
                            className="sr-only"
                          >
                            Previous
                          </span>
                        </a>
                      </PaginationLink>
                    </li>
                  </PaginationItem>
                  <PaginationItem
                    className=""
                    key="1"
                    tag="li"
                  >
                    <li
                      className="page-item"
                    >
                      <PaginationLink
                        onClick={[Function]}
                        tag="a"
                      >
                        <a
                          className="page-link"
                          onClick={[Function]}
                        >
                          1
                        </a>
                      </PaginationLink>
                    </li>
                  </PaginationItem>
                  <PaginationItem
                    className="active"
                    key="2"
                    tag="li"
                  >
                    <li
                      className="active page-item"
                    >
                      <PaginationLink
                        onClick={[Function]}
                        tag="a"
                      >
                        <a
                          className="page-link"
                          onClick={[Function]}
                        >
                          2
                        </a>
                      </PaginationLink>
                    </li>
                  </PaginationItem>
                  <PaginationItem
                    className=""
                    key="3"
                    tag="li"
                  >
                    <li
                      className="page-item"
                    >
                      <PaginationLink
                        onClick={[Function]}
                        tag="a"
                      >
                        <a
                          className="page-link"
                          onClick={[Function]}
                        >
                          3
                        </a>
                      </PaginationLink>
                    </li>
                  </PaginationItem>
                  <PaginationItem
                    className=""
                    tag="li"
                  >
                    <li
                      className="page-item"
                    >
                      <PaginationLink
                        onClick={[Function]}
                        tag="a"
                      >
                        <a
                          className="page-link"
                          onClick={[Function]}
                        >
                          <span
                            aria-hidden="true"
                          >
                            »
                          </span>
                          <span
                            className="sr-only"
                          >
                            Next
                          </span>
                        </a>
                      </PaginationLink>
                    </li>
                  </PaginationItem>
                  <PaginationItem
                    className=""
                    tag="li"
                  >
                    <li
                      className="page-item"
                    >
                      <PaginationLink
                        onClick={[Function]}
                        tag="a"
                      >
                        <a
                          className="page-link"
                          onClick={[Function]}
                        >
                          Last
                        </a>
                      </PaginationLink>
                    </li>
                  </PaginationItem>
                </ul>
              </Pagination>
            </nav>
          </div>
        </Pagination>
      </div>
    </RecipeItems>
  </div>
</Home>
`;
